### ISSUE TITLE
[Feature] Add daily # of tests performed on daily number of positives confirmed

### ISSUE THREAD

**Issue by anelda** [dsfsi/covid19za#17]
Hi Vukosi,

I'm thinking it may be useful to display the number of tests done against the number of cases confirmed on the barchart on page 2 of the dashboard. Or showing positives vs negatives. It will give us an idea of how many tests are essentially wasted because we're testing unnecessary due to not being able to prioritize patients and highlighting the symptoms that are required for testing? But at the same time realising that people will be tested if they've been in touch with positives even if they aren't displaying symptoms. 

Not wasting tests on the wrong patients seems like an important aspect of successful response?

Thanks,
Anelda

**Comment by vukosim** [comment]
Can you maybe provide a jupyter notebook with the visualisation. Just clone the repo and do a pull request.

**Comment by freever** [comment]
@anelda I also think number of tests conducted per day would be valuable, but for a different reason. If there are 10 new cases today out of 20 tests conducted, and 50 new cases tomorrow out of 1000 tests conducted, then I won't be alarmed about the spike in cases because I know it comes from ramping up our testing efforts. 

Either way, tests condicted per day would be valuable information to have if you can add it. 

Thanks for the great work!

**Comment by vukosim** [comment]
Thanks both. There is a csv file with the cumulative tests. If you can provide a jupyter notebook or graph with your idea here, we can test it out @anelda @freever

**Comment by freever** [comment]
@vukosim what is a jupyter notebook? do you just want to see a visual mockup of how it could look?

**Comment by anelda** [comment]
Hi both, sorry yes, I cloned the repo already and will try to do the plot this morning. @freever if you have an idea of how the plot should look, let me know. I have experience with Jupyter and can write the code (python) - do you have experience with Python? I got a bit sidetracked when I started thinking about what the best  way would be to display it. Thinking a stacked column showing negatives vs positives up to the number of tests done. @freever I agree with the reason for wanting to see the plot like this. I thought about that as well when looking at the data

**Comment by freever** [comment]
@anelda A stacked graph sounds perfect. I don't know anything about python, I will leave it to the experts. Just wanted to ask for the data :)

**Comment by anelda** [comment]
I have two ways of showing this data now, but there are some challenges to show this daily:

- tests take 2 days (+-) before results are available, so not sure how valid it is to do [tests - positives] = negatives per day (see also issue #26)
- there are days when we don't have data for how many tests were perform - we can extrapolate the data, but it won't be true (e.g. 9 March - no info about tests, but 4 positives confirmed). This also have implications for using the above method for calculating negatives per day
- currently the number of tests are way more than the number of positives, so I tried to do a plot with two y-scales, but the graph really isn't useful to tell a story
- I tried line chart combined with bar plot, but no good. Stacked bar isn't good either. Neither is grouped barplot.

If anyone have a suggestion for a good way to show the data, please feel free to let me know and I can try to code it?

**Comment by freever** [comment]
Maybe it would be best just to show the number of reported tests on their own. If we don't have a reliable source for both number of tests and number of positive tests then it won't make sense to show them together. 

It might be nice to have 2 axes - one for tests conducted per day, and one for cumlative tests conducted up to that day - but I guess it depends on how the data actually looks when plotted.

**Comment by anelda** [comment]
Found this graph for showing # of positives vs # tests - https://ourworldindata.org/exports/tests-vs-confirmed-cases-covid-19_v7_850x600.svg but we don't have sufficient information to do something similar. If we had info about how many people were tested per province, that might be useful, but as far as I Googled, that level of info is not available.

**Comment by anelda** [comment]
@vukosim the Hong Kong dashboard is very clean - not showing graph of number of tests done vs positive/negative, but showing the numbers some other issues are also asking for. How about doing it like this rather than showing another plot. At least until we have more data points that will look better on a plot? https://chp-dashboard.geodata.gov.hk/covid-19/en.html

**Comment by devontighe** [comment]
@anelda I would caution referred to tests being "wasted". I 100% agree with you that we want testing to be prioritized amongst those who need it most. But ultimately the more people who are tested, the better. South Korea has taught us that. Maybe something like this could work?  (or hold off on including % recovered to start as that % recovered would overlap with % negative)

![SA_Covid19_chart](https://user-images.githubusercontent.com/25172032/77152584-eeaf5100-6aa0-11ea-8234-b3ad0a6c0081.jpeg)

**Comment by anelda** [comment]
@devontighe thanks so much for pointing out the negative connotation to using the word 'wasted' here. Bad choice of words, sorry! Yes, totally agree. More tests are better. Think I was concerned that random people will go for testing which will make it impossible for people who had high chance of really being infected tested, but sounds like that shouldn't happen given the criteria for testing?

Thanks so much for your data viz suggestion. I think the challenge with the data is several fold (but maybe I'm making a bigger deal of this than necessary):

- the lag between date test is conducted and date result become available (48 - 72 hours lag) - not sure which number is reported currently by the DoH - I suppose it is tests with outcome, not samples collected for testing or tests in other phase than concluded?
- in the way that you're presenting it here, it makes it look as if the number of people recovering (or passing away)  is linked at any time to the tests done on that day. Maybe that's why people are rather showing cumulative data so that there is no perception that the data is linked when it isn't? Do you think that's a problem? Or do you mean to present the data as cumulative data so day 5 refers to the numbers after 5 days not on day 5?
- is your 100% here the total population of SA? I like the idea, but we'll have to consider the scale to be able to see the smaller values vs the huge population size e.g. 150 positives as percentage of 59 mil will hardly be visible. What would you propose to get around that? 
- I'm wondering if some people may have more than one test (especially over the course of the pandemic or when testing negative but symptoms getting more severe after a few days)? That might make our 100% a problem if it is population size?

Am I overthinking this?

**Comment by anelda** [comment]
Yip, so just to re-iterated what you said above. For someone to be declared 'recovered' they have to have 2 negative tests. I can't find a source for SA to confirm that's how recovered cases are classified here. Only found link to US CDC - https://www.cdc.gov/coronavirus/2019-ncov/healthcare-facilities/hcp-return-work.html.

What does it mean for how the data is shown? In my opinion then negative tests don't mean anything unless we have more information about them? E.g. was it a follow-up test after getting better or to diagnose in the first place

**Comment by elolelo** [comment]
@anelda @devontighe @freever Hi all, are you winning  with getting more information about negative cases? Is there any source from where data about follow-up tests can be recieved? And which data viz are you going to use ? my assumption is that  this feature is still going to be implemented, or is it not?

**Comment by anelda** [comment]
> @anelda @devontighe @freever Hi all, are you winning with getting more information about negative cases? Is there any source from where data about follow-up tests can be recieved? And which data viz are you going to use ? my assumption is that this feature is still going to be implemented, or is it not?

@elolelo just not sure that there's a lot of value in plotting this right now. Not until we have more information about the tests e.g. same person being tested several times, etc. I guess we may not be able to get that kind of information so maybe okay to close this? What do you think @freever @devontighe @vukosim? We can always revisit when more information is available?

**Comment by freever** [comment]
I dunno, this is not really my field, but I had a look at the data and generated this graph to answer a question I have about whether the recent spike in cases has been caused by a rapid increase in the number of tests conducted. Spoiler: no, it hasn't. I think there is value in tracking this graph over time:

![cases and tests](https://user-images.githubusercontent.com/2269624/77536501-150c2c80-6ea5-11ea-80a0-72487b3058fe.png)

**Comment by vukosim** [comment]
@anelda on the dashboard this is now available [has been for a while now]. I have been tardy
